doc/packaging.md

```diff
@@ -0,0 +1,334 @@
+
+## Updating from `iotedge` to `aziot-edge`
+
+```sh
+apt remove iotedge libiothsm-std
+
+apt install aziot-edge
+
+sudo iotedge config import
+```
+
+The `iotedge config import` can be used to generate the `/etc/aziot/config.toml` based on the old `/etc/iotedge/config.yaml` left behind after uninstalling versions 1.1 or earlier of IoT Edge. This assumes that the `--purge` flag is not used to remove packages as that would remove the old configuration.
+
+We do not support simultaneously running the services from both the old and new packages. They would step over each other trying to provision the device and manage Docker modules. We enforce mutual exclusivity between the packages by having them conflict with each other. This causes the distribution's package manager (e.g `apt` or `apt-get`) to not allow them both to be installed at the same time. This does not work if `dpkg` is used directly. Instead, the existing `iotedge` and `libiothsm-std` packages must be removed before installing the `aziot-edge` package (or even the `aziot-identity-service` package).
+
+### Importing Configuration
+
+This is provided as a convenience to ease the transition from v1.1 to v1.2.
+
+```sh
+iotedge config import
+```
+
+Notes on behavior:
+
+- The device provisioning method is parsed from `/etc/iotedge/config.yaml` and translated into the provisioning information in `/etc/aziot/config.toml`.
+
+- User-provided certificates like device ID, device CA and trust bundle, and their corresponding private key files, will be added as preloaded keys and certs in `config.toml`. The files themselves will not be moved, because they are managed by the user rather than belonging in our services' directories.
+
+  This assumes that Microsoft's implementation of `libiothsm-std` is being used where the certs and keys are stored as files on disk. This is a reasonable assumption since we are not aware of any external Edge customers that have written their own `libiothsm-std` implementations which store keys and certs differently.
+
+- The master identity key and master encryption key are two symmetric keys dynamically generated by `iotedged` for internal use. They will not me be imported into the new services.
+
+  Note that this means Edge Agent, Edge Hub and other modules will not be able to decrypt any data when running against the new services that they previously encrypted using the workload API with the old service. This is not a problem for Edge Agent and Edge Hub, in the sense that they ignore the undecryptable data and start with empty storage, so they recover at the cost of data loss.
+
+- Certs like workload CA and module server certs that are created dynamically by `iotedged`, and can be regenerated trivially without any problems, will not be imported into the new services.
```

Comment:
The reasons are different for them so it would be good to separate them:

```
- The workload CA cert is no longer used with aziot-edged and will not be imported.

- Module identity certs and module server certs will not be imported. Modules will get new certs signed directly by the Edge CA (nee the device CA) when they request them.
```

Reply:
Split as recommended